frontend/projects/ui/src/app/guards/auth.guard.ts

```diff
@@ -1,38 +1,29 @@
 import { Injectable } from '@angular/core'
-import { CanActivate, Router, CanActivateChild } from '@angular/router'
-import { tap } from 'rxjs/operators'
-import { AuthState, AuthService } from '../services/auth.service'
+import { CanActivate, Router, CanActivateChild, UrlTree } from '@angular/router'
+import { map } from 'rxjs/operators'
+import { AuthService } from '../services/auth.service'
+import { Observable } from 'rxjs'
 
 @Injectable({
   providedIn: 'root',
 })
 export class AuthGuard implements CanActivate, CanActivateChild {
-  authState: AuthState
-
-  constructor (
+  constructor(
     private readonly authService: AuthService,
     private readonly router: Router,
-  ) {
-    this.authService.watch$()
-    .pipe(
-      tap(auth => this.authState = auth),
-    ).subscribe()
-  }
+  ) {}
 
-  canActivate (): boolean {
+  canActivate(): Observable<boolean | UrlTree> {
```

Comment:
:eye: That's a weird type

Reply:
Angular guards can return new `UrlTree` to be redirected to or `true` to let router know the navigation is allowed by the guard.

Reply:
This is why I ask questions, thank you.